src/android/jni_domain_tests.rs

```diff
@@ -0,0 +1,229 @@
+    _: JClass,
+) -> jobject {
+    let alert1 = create_test_alert("123", 131321);
+    let alert2 = create_test_alert("343356", 32516899200);
+    let jobject1 = alert_to_jobject(alert1, &env);
+    let jobject2 = alert_to_jobject(alert2, &env);
+
+    let array: jobjectArray = env
+        .new_object_array(2, "org/coepi/android/api/JniAlert", jobject1)
+        .unwrap();
+    env.set_object_array_element(array, 0, jobject1).unwrap();
+    env.set_object_array_element(array, 1, jobject2).unwrap();
+
+    jni_obj_result(
+        1,
+        None,
+        JObject::from(array),
+        "org/coepi/android/api/JniAlertsArrayResult",
+        "[Lorg/coepi/android/api/JniAlert;",
+        &env,
+    )
+}
+
+fn create_test_alert(id: &str, report_time: u64) -> Alert {
+    let report = PublicReport {
+        report_time: UnixTime { value: report_time },
+        earliest_symptom_time: UserInput::Some(UnixTime { value: 1590356601 }),
+        fever_severity: FeverSeverity::Mild,
+        cough_severity: CoughSeverity::Dry,
+        breathlessness: true,
+        muscle_aches: true,
+        loss_smell_or_taste: false,
+        diarrhea: false,
+        runny_nose: true,
+        other: false,
+        no_symptoms: true,
+    };
+
+    Alert {
+        id: id.to_owned(),
+        report,
+        contact_time: 1592567315,
+    }
+}
+
+fn alert_to_jobject(alert: Alert, env: &JNIEnv) -> jobject {
+    let jni_public_report_class = env
+        .find_class("org/coepi/android/api/JniPublicReport")
+        .unwrap();
+
+    let report_time_j_value = JValue::from(alert.report.report_time.value as i64);
+
+    let earliest_time = match &alert.report.earliest_symptom_time {
+        UserInput::Some(time) => time.value as i64,
+        UserInput::None => -1,
+    };
+    let earliest_time_j_value = JValue::from(earliest_time);
+
+    let fever_severity = match &alert.report.fever_severity {
+        FeverSeverity::None => 0,
+        FeverSeverity::Mild => 1,
+        FeverSeverity::Serious => 2,
+    };
+    let fever_severity_j_value = JValue::from(fever_severity);
+
+    let cough_severity = match &alert.report.cough_severity {
+        CoughSeverity::None => 0,
+        CoughSeverity::Existing => 1,
+        CoughSeverity::Wet => 2,
+        CoughSeverity::Dry => 3,
+    };
+    let cough_severity_j_value = JValue::from(cough_severity);
+
+    let breathlessness_j_value = JValue::from(alert.report.breathlessness);
+    let muscle_aches_j_value = JValue::from(alert.report.muscle_aches);
+    let loss_smell_or_taste_j_value = JValue::from(alert.report.loss_smell_or_taste);
+    let diarrhea_j_value = JValue::from(alert.report.diarrhea);
+    let runny_nose_j_value = JValue::from(alert.report.runny_nose);
+    let other_j_value = JValue::from(alert.report.other);
+    let no_symptoms_j_value = JValue::from(alert.report.no_symptoms);
+
+    let jni_public_report_obj = env.new_object(
+        jni_public_report_class,
+        "(JJIIZZZZZZZ)V",
```

Comment:
@i-schuetz  What is this?

Reply:
The JNI signature for the public report constructor https://docs.oracle.com/javase/7/docs/technotes/guides/jni/spec/types.html

Reply:
(The Android specific one that uses only primitives, that is)